src/lib/SectionTitle.svelte

```diff
@@ -1,13 +1,30 @@
-<script>
-  let id = "temp-" + Math.random()
-  let container
+<script lang="ts">
+  import { getContext, onMount, setContext } from "svelte"
+  import { tableOfContents } from "../routes/store"
 
-  $: if (container) {
-    id = container.innerText.trim().replace(" ", "-").toLowerCase()
+  let id: string
+  export let level = 1
+  export let title
+
+  $: if (title) {
+    id = title.trim().replaceAll(" ", "-").toLowerCase()
   }
+  const context = getContext("TableOfContents")
+  onMount(() => {
+    if (context) {
+      if (title && id) {
+        context.addEntry({
+          title: title,
+          level: level,
+          id: id,
+          active: $tableOfContents.length > 0 ? false : true,
```

Comment:
This seems wrong and confusing, and why is the children accessing the store? If any logic like this needs to exist it should be in the parent.